client/src/containers/SubmitCode.js

```diff
@@ -1,62 +1,93 @@
 import React, { PropTypes, Component } from 'react';
-import { submitCode } from '../actions/code';
+import { getUserInfo } from '../actions/user.js';
 import { connect } from 'react-redux';
 
 class SubmitCode extends Component {
 	constructor(props) {
 		super(props);
-		this.state = {
-			value: '(e.g. CS 242 - Chess)', 
-			language: {value: 'Java'}
-		};
 		// React components in ES6 no long autobinds this to nonReact methods
-		this.handleChange = this.handleChange.bind(this);
-		this.handleSelect = this.handleSelect.bind(this);
-		this.handleSubmit = this.handleSubmit.bind(this);
+		this.handleTitleChange = this.handleTitleChange.bind(this);
+		this.handleLanguageSelect = this.handleLanguageSelect.bind(this);
+		this.handleContentChange = this.handleContentChange.bind(this);
+		this.routeToSubmitCodeSmells = this.routeToSubmitCodeSmells.bind(this);
+		this.state = {
+			content: ""
+		}
+	}
+
+	componentDidMount() {
+        const { dispatch } = this.props;
+        dispatch(getUserInfo());
+    }
+
+	handleTitleChange(evt) {
+		const { code } = this.props;
+		code.title = evt.target.value;
```

Comment:
I don't think this should be a prop. It should be a local state. I think the style is that you don't change the prop values cause they're supposed to be const or something.